main.js

```diff
@@ -1,107 +1,290 @@
         this.openedBlocks = [];
     }
-    Board.prototype.draw = function () {
-        var _this = this;
-        for (var i = 0; i < this.blocks.length; i++) {
-            var row = document.createElement('div');
-            row.classList.add('row');
-            var _loop_1 = function (j) {
-                var div = document.createElement('div');
-                var block = this_1.blocks[i][j];
-                var attributes = block.getAttributes();
-                div.style.width = attributes.width.toString().concat("px");
-                div.style.height = attributes.height.toString().concat("px");
-                div.classList.add('block');
-                div.style.backgroundColor = 'black';
-                block.setDivElement(div);
-                div.addEventListener('click', function () { return _this.open(block); });
-                row === null || row === void 0 ? void 0 : row.appendChild(div);
-            };
-            var this_1 = this;
-            for (var j = 0; j < this.blocks.length; j++) {
-                _loop_1(j);
+    shuffleBlocks(blocksArray) {
+        let currentIndex = blocksArray.length, randomIndex;
+        while (currentIndex !== 0) {
+            randomIndex = Math.floor(Math.random() * currentIndex);
+            currentIndex--;
+            [blocksArray[currentIndex], blocksArray[randomIndex]] = [
+                blocksArray[randomIndex],
+                blocksArray[currentIndex],
+            ];
+        }
+        return blocksArray;
+    }
+    draw() {
+        for (let i = 0; i < this.blocks.length; i++) {
+            const row = document.createElement("div");
+            row.classList.add("row");
+            for (let j = 0; j < this.blocks.length; j++) {
+                // TODO here we need to create a element creator class
+                // to create the will be appended and set in the block
+                //const div = document.createElement("div");
+                const block = this.blocks[i][j];
+                const attributes = block.getAttributes();
+                const div = new ElementCreator("div", attributes.width.toString(), attributes.height.toString(), undefined, { backgroundColor: "black" }, // Style property
+                "block").createElement();
+                //div.style.width = attributes.width.toString().concat("px");
+                //div.style.height = attributes.height.toString().concat("px");
+                //div.classList.add("block");
+                //div.style.backgroundColor = "black";
+                block.setElement(div);
+                const openBlockFunction = () => this.open(block);
+                div.addEventListener("click", openBlockFunction);
+                block.setOpenFunction(openBlockFunction);
+                row?.appendChild(div);
             }
             this.container.appendChild(row);
         }
-    };
-    Board.prototype.open = function (block) {
-        var _this = this;
+    }
+    open(block) {
```

Comment:
You should revise the open function. I won't think it will work as expected. I should not be able to click the block again if I press it multiple times during the opening animation. Also, a lot of crowded code here extract and move to function.